storage/filesystem/object.go

```diff
@@ -402,6 +401,14 @@ func (s *ObjectStorage) getFromUnpacked(h plumbing.Hash) (obj plumbing.EncodedOb
 		return nil, err
 	}
 
+	if size > packfile.LargeObjectThreshold {
+		obj = dotgit.NewEncodedObject(s.dir, h, t, size)
+		s.objectCache.Put(obj)
```

Comment:
Should we put the object in cache when is bigger than `packfile.LargeObjectThreshold`?

Reply:
So the object that is cached here is not a MemoryObject  but a dotgit.NewEncodedObject - which is a thin wrapper around the reader. (see above) The only ways this would be incorrect is if there is a repack and this object is packed away or this object is gc'd away.

I'm not sure it's correct though - if you think it's better not to put it in the cache then happy to remove.

Reply:
It may not occupy memory but I believe the cache will use the size of the object to evict previous objects if there's no more space.

I may me incorrect. It'a a while I don't work with this codebase.

Reply:
ah you're right let me remove this.

```suggestion
```